sklearn/utils/fixes.py

```diff
@@ -335,3 +335,20 @@ def __call__(self, *args, **keywords):
             return self.func(*args, **kwargs)
 else:
     from functools import partial
+
+
+if np_version < (1, 6, 2):
+    # Allow bincount to accept empty arrays
+    # https://github.com/numpy/numpy/commit/40f0844846a9d7665616b142407a3d74cb65a040
+    def bincount(x, weights=None, minlength=None):
+        x = np.asarray(x, dtype=np.intp)
+        if x.size > 0:
+            return np.bincount(x, weights, minlength)
+        else:
+            if minlength is None:
+                minlength = 0
+            minlength = np.asscalar(np.asarray(minlength, dtype=np.intp))
+            return np.zeros(minlength, dtype=np.intp)
```

Comment:
Does what I wrote not work? 



On Feb 1, 2015, 17:54, at 17:54, "Loïc Estève" <notifications@github.com> wrote:
>> @@ -335,3 +335,20 @@ def __call__(self, *args, **keywords):
>>              return self.func(*args, **kwargs)
>>  else:
>>      from functools import partial
>> +
>> +
>> +if np_version < (1, 6, 2):
>> +    # Allow bincount to accept empty arrays
>> +    #
>https://github.com/numpy/numpy/commit/40f0844846a9d7665616b142407a3d74cb65a040
>> +    def bincount(x, weights=None, minlength=None):
>> +        x = np.asarray(x, dtype=np.intp)
>> +        if x.size > 0:
>> +            return np.bincount(x, weights, minlength)
>> +        else:
>> +            if minlength is None:
>> +                minlength = 0
>> +            minlength = np.asscalar(np.asarray(minlength,
>dtype=np.intp))
>> +            return np.zeros(minlength, dtype=np.intp)
>
>> np.asscalar(minlength).astype(np.intp)?
>
>I am afraid np.asarray(minlength) raises an exception when minlength is
>an int.
>
>My understanding is the C code is that you want to allow minlength to
>be an int or a scalar array with an int (int or anything castable to an
>int actually).
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/scikit-learn/scikit-learn/pull/4193/files#r23898980